.github/workflows/ci-go.yml

```diff
@@ -0,0 +1,46 @@
+# Continuous integration handling for Go
+name: ci-go
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/test.yml
+      - .go-version
+      - go.mod
+      - '**.go'
+
+permissions:
+  contents: read
+
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: go-version
+        # Reference: https://github.com/actions/setup-go/issues/23
+        run: echo "::set-output name=version::$(cat ./.go-version)"
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ steps.go-version.outputs.version }}
+      - run: go mod download
+      - uses: golangci/golangci-lint-action@v2
+        with:
+          skip-go-installation: true
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: [ '1.17', '1.16' ]
```

Comment:
Nit: renaming this to `go-version` may improve readability in the future for any workflows that have multiple long-form steps. Not an issue given the conciseness of this job, however.

Reply:
I definitely appreciate this change and will make it! 😄